src/hiveboard_bridge/src/TCPServer.cpp

```diff
@@ -44,33 +41,30 @@ void TCPServer::listen() {
                                (socklen_t*)&m_addressLength)) < 0) {
         ROS_ERROR("TCP server accept failed");
     }
-
-    m_monitor.addConn(m_clientFd);
 }
 
-int TCPServer::read(char* buff, const uint16_t length, bool blocking) {
-    if (blocking) {
-        if (!m_monitor.waitIncoming(m_clientFd)) {
-            ROS_ERROR("TCP server error waiting for incoming data");
-        }
-    }
-
-    int nbBytesRecieved = ::recv(m_clientFd, m_buffer, length, 0);
+bool TCPServer::receive(uint8_t* data, uint16_t length) {
+    int nbBytesReceived = ::recv(m_clientFd, data, length, MSG_WAITALL);
 
-    if (blocking && nbBytesRecieved == 0) { // Client disconnected
+    if (nbBytesReceived == 0) { // Client disconnected
+        ROS_WARN("TCP client disconnected");
         close();
     }
 
-    std::memcpy(buff, m_buffer, length);
+    if (nbBytesReceived == length)
+        return true;
 
-    return nbBytesRecieved;
+    return false;
```

Comment:
No need for two returns at line 55 and 57, you can merge them via
```suggestion
    return nbBytesReceived == length;
```